baseapp/baseapp.go

```diff
@@ -997,13 +997,28 @@ func (h DefaultProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHand
 // DefaultPrepareProposal. It is very important that the same validation logic
 // is used in both steps, and applications must ensure that this is the case in
 // non-default handlers.
+//
+// Additional to the above mentioned conditions, total gas wanted by all
+// transactions in the proposal must not be more gas than the block gas limit.
+// sum_tx_gas_wanted <= block_gas_limit
 func (h DefaultProposalHandler) ProcessProposalHandler() sdk.ProcessProposalHandler {
 	return func(ctx sdk.Context, req abci.RequestProcessProposal) abci.ResponseProcessProposal {
+		var gasWanted uint64
 		for _, txBytes := range req.Txs {
-			_, err := h.txVerifier.ProcessProposalVerifyTx(txBytes)
+			tx, err := h.txVerifier.ProcessProposalVerifyTx(txBytes)
 			if err != nil {
 				return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
 			}
+
+			feeTx, ok := tx.(sdk.FeeTx)
+			if !ok {
+				return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
+			}
+			gasWanted += feeTx.GetGas()
+		}
+
+		if gasWanted > ctx.BlockGasMeter().Limit() {
+			return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
```

Comment:
Why do this as a bolt-on in the default handler instead of just adding a new AnteHandler decorator or even update an existing one?

Reply:
https://github.com/cosmos/cosmos-sdk/issues/14540#issuecomment-1384634585
You mentioned we could add this to our default handler.
Also does "new AnteHandler decorator" mean creating a new one in the x/auth/ante ? If yes, How do I add that handler to  ProcessProposerHandler?
Or do you want me edit or create a new fn like `ProcessProposalVerifyTx`?

Reply:
Yeah, so let me rephrase. The provided `ProcessProposalVerifyTx` to apps is essentially a utility function that wraps `runTx` with mode == processProposal, which itself is just a call to the anteHandler.

In other words, `ProcessProposalVerifyTx` is just running the AnteHandler chain.

So my point is, why add this logic as a hard-coded extension, instead of just adding to the AnteHandler chain? It seems much cleaner to me that way.